portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -92,5 +82,38 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
+
+  /**
+   * @return the requested number of comments to be shown, or, if none is requested, 
+   * the total number of comments in the comment history. 
+   */
+  private int getCommentLimit(HttpServletRequest request, int total) {
+    String requestedCommentLimit = getParameter(request, "comment-limit", "All");
+    int commentLimit;
+    if(requestedCommentLimit.equals("All")){
+      commentLimit = total;
+    } else {
+      commentLimit = Integer.parseInt(requestedCommentLimit);
+    }
+    return commentLimit; 
+  }
+
+  /**
+   * @return An ArrayList containing the commentHistory up to the given commentLimit
+   */
+  private ArrayList<Comment> getCommentHistory(PreparedQuery storedComments, int commentLimit) {
+    ArrayList<Comment> commentHistory = new ArrayList<Comment>(); 
+    for(Entity comment : storedComments.asIterable()){
```

Comment:
A while loop might be easier to follow here. Basically that's because if you see `while(i < commentLimit) {` you already know that you will iterate `commentLimit` times. If you do a for loop with a break instead, it looks like you will iterated `len(storedComments)` times, then as you read you realize that this isn't the case.